### PR TITLE
[BugFix] Fix prune map_values bug

### PR DESCRIPTION
**Pull request by Seaven**
Fixes #issue

When `map_keys` and `map_values` appear at the same time, the values is prune by mistake

```
MySQL test> select  map_values(col_map) from test_semi_struct_array_functions;
+---------------------------------+
| map_values(col_map)             |
+---------------------------------+
| [{"2023-01-01":{1.11:1.11100}}] |
+---------------------------------+
1 row in set
Time: 0.038s
MySQL test> select map_keys(col_map), map_values(col_map) from test_semi_struct_array_functions;
+-------------------+---------------------+
| map_keys(col_map) | map_values(col_map) |
+-------------------+---------------------+
| [1]               | [null]              |
+-------------------+---------------------+
1 row in set
Time: 0.037s
MySQL test>
```

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/PruneSubfieldRule.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldRule.java [hunk 1]
```diff
@@ -49,6 +49,9 @@
 import java.util.stream.Collectors;
 
 public class PruneSubfieldRule implements TreeRewriteRule {
+    private static final List<String> SUPPORT_FUNCTIONS = ImmutableList.<String>builder()
+            .add(FunctionSet.MAP_KEYS, FunctionSet.MAP_SIZE)
```
Comment (fzhedu) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldRule.java:53]: how about these two appear together? add some tests pls

**Review by packy92 — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldRule.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PruneSubfieldRule.java:53]: already have

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/tree/PruneSubfieldRule.java|4|4|100.00%|[]|

**Review by fzhedu — APPROVED** [review]
LGTM

**Review by stdpain — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**